README.md

```diff
@@ -2,19 +2,71 @@
 
 Let your software roar!
 
-## Architecture
-Tuesday concept:
+## Assumptions and Requirements
+Let's start with an analogy: the profiled program is like a mechanical gearbox with many cogs that each can make a sound. The sound a cog makes depends on 1) how fast it spins 2) how much power it transmits 3) the shape/material/kind of the cog.
+
+Acoustic profiler makes it fairly easy to
+1. implement probes for various types of cogs
+2. vary sound effects given to individual cogs
+3. hear cogs spinning in remote machine (host)
+
+This necessitates an interface somewhere between the probes and the sound synthesis so that they can be mix-and-matched. This interface also works over network to satisfy 3.
+
+## Interface
+
+On abstract level, the interface is just a stream of Events where event is
+
+```rust
+struct Event {
+    type: EventType
+    // only values between [0, 1]
+    quality: f32
+}
+
+enum EventType {
+    ActorMessageSent,
+    FilesystemWriteOrRead
+    ...
+}
+```
+
+The frequency of events encodes how fast a cog spins and the quality encodes the power transmitted through it. (We can add more dimensions later.) We also assume that
```

Comment:
Instead of relying on the event frequency to be transmitted over the wire we may instead make the probes to send messages *regularly* (Eg. every 0.1s) and encode the actual event frequency as an *attribute* of the regularly sent message.

I.e. a message then would be something like
```rust
Message {
    event_type: EventType,
    // occurences of the events in the last 0.1 s
    occurences: u32,
    // the sum of qualities
    quality: f32,
}
```

This could
1) save bandwith
2) curcumvent timing/ordering/buffering issues in the network layer

OTOH we'd also loose some resolution that could be used for sound synthesis.

Reply:
I like starting with the simplest (as in the original proposal) and evolve over time. We can revisit the idea of `occurrences` and `frequency` later.

A thought for later: I think I'm starting to prefer making probes as dumb as possible and let the server handle most of the complexities. 
* The server knows more about kinds of sound sources it can play and other events from concurrent probes, which may influence how to interpret a particular series of events. Disk reads could be thought of as a changing frequency value, or a series of discrete events.
* We can record incoming probe events and play back on the server side to quickly iterate on aggregation and synthesis.
* We probably want to encode timestamp in the event struct as well, so we are not influenced by the network conditions. It allows us to send `Vec<Event>` to reduce serialization and network overhead.